launch/launch/substitutions/path_join_substitution.py

```diff
@@ -44,3 +44,6 @@ def perform(self, context: LaunchContext) -> Text:
         """Perform the substitution by retrieving the local variable."""
         performed_substitutions = [sub.perform(context) for sub in self.__substitutions]
         return os.path.join(*performed_substitutions)
+
+    def __repr__(self):
+        return self.describe()
```

Comment:
I know this isn't an issue as-is, but, since [`Substitution.describe()` calls `repr(self)`](https://github.com/ros2/launch/blob/a9258c3e23238be022869edca19c67d45a932d6e/launch/launch/substitution.py#L34), this could be kind of dangerous if `PathJoinSubstitution.describe()` is removed. I would move the implementation of `PathJoinSubstitution.describe()` to `PathJoinSubstitution.__repr__()` (here). I would then probably remove `PathJoinSubstitution.describe()`.

Reply:
Yes that make sense. I did it this way simply to minimize the changes in this PR.
I've since uncovered this older issue about refactoring the introspection system: #103. Reading that made me wonder why a separate `describe()` is needed when it seems to serve the same purpose as `__repr__()`.
Do you want me to make this change now or just keep in mind if/when I decide to help with fleshing out #103?

Reply:
I would just make the change now, and then later potentially re-consider everything as part of #103.